### DISCUSSION TITLE
How Can I colored message?

### DISCUSSION THREAD

**Discussion by og900aero** [gokcehan/lf#1355] [Q&A] [answered]
I use custom rule:

```
cmd totalsize &{{
	lf -remote "send $id set ruler progress:selection:ind:df:lf_user_selected_size"
	s=$(du -ch $fx | tail -n 1 | cut -f 1)
	lf -remote "send $id set user_selected_size $s"
}}
```
How can I write in color? I think the remote send must be parameterized somehow, the only question is how?

**Comment by joelim-work** [comment]
You can include color escape sequences when setting a user variable, just change the last line to this:

```shell
lf -remote "send $id set user_selected_size \"\033[32m$s\033[0m\""
```

Note that the inner pair of double quotes has to be escaped, and is required otherwise `\033` won't be interpreted as an escape character.

---

However the problem is that currently `lf` doesn't support escape sequences when displaying the `ruler`, you can look at https://github.com/gokcehan/lf/pull/1257#discussion_r1199547219 for more details. We could just delete the following code, and then the `ruler` should display the color escape sequence properly: https://github.com/gokcehan/lf/blob/2620f492c27a24205ace5f5e791a0179f0489f88/ui.go#L811-L812

I'm not sure what is the best thing to do in this case. I think it would be nicer if the colors could be handled inside the `ruler` option instead so that internal fields like `ind` and `df` could be colored too, but currently this isn't supported.

It might be better to raise this as an actual issue.

**Reply by og900aero** [reply]
Unfortunately it doesn't work. See in the photo:

[Link](https://www.kepfeltoltes.eu/images/2023/07/27/805problem.jpg)

**Reply by joelim-work** [reply, marked as answer]
Yeah that's what I was referring to when I said escape sequences don't work when displaying the ruler, instead the escape character is displayed as `\033` in reverse mode. If you delete that bit of code and then recompile `lf` then it should work.

So the reasons for displaying escape sequences instead of processing them was:

1. To be able to display the literal values of options like `promptfmt` in the ruler
2. Security reasons (?)

Even now, I'm not sure about that decision.

**Reply by og900aero** [reply]
All right. So until I know for sure if this will be supported or not, I won't use it.

**Reply by joelim-work** [reply]
Well if it interests you, I wrote up an idea for implementing a new `rulerfmt` option, see #1358 for details.

It's just a proof-of-concept though, so don't get too excited.

**Reply by og900aero** [reply]
Okay, thx.
